pkg/sql/sqlutil/internal_executor.go

```diff
@@ -158,6 +167,18 @@ type InternalExecutor interface {
 	WithSyntheticDescriptors(
 		descs []catalog.Descriptor, run func() error,
 	) error
+
+	// SetExtraTxnState is to set the extra txn state for an internal executor.
+	// It should only be called if the internal executor is used to run sql
+	// sql statements under a planner context.
+	SetExtraTxnState(extraTxnState ExtraTxnStateUnderPlanner)
+
+	// SetDescsCollection is to set the descriptor collection used by the internal
+	// executor.
+	SetDescsCollection(txn *kv.Txn, collection descsinterface.DescriptorCollection)
+
+	// MakeDescsCollection creates the internal executor's own descriptor collection.
+	MakeDescsCollection(ctx context.Context, sd *sessiondata.SessionData) descsinterface.DescriptorCollection
```

Comment:
This is pretty gross as far as I'm concerned. Is there a way we can push this into the object or methods which constructs an internal executor such that the internal executor has a tight coupling to these?